util/chunk/column.go

```diff
@@ -246,6 +241,83 @@ const (
 	sizeMyDecimal = int(unsafe.Sizeof(types.MyDecimal{}))
 )
 
+// preAlloc allocates space for a fixed-length-type slice and resets all slots to null.
+func (c *Column) preAlloc(length, typeSize int) {
+	nData := length * typeSize
+	if len(c.data) >= nData {
+		c.data = c.data[:nData]
+	} else {
+		c.data = make([]byte, nData)
+	}
+
+	nBitmap := (length + 7) >> 3
+	if len(c.nullBitmap) >= nBitmap {
+		c.nullBitmap = c.nullBitmap[:nBitmap]
+		for i := range c.nullBitmap {
+			// resets all slots to null.
+			c.nullBitmap[i] = 255
+		}
+	} else {
+		c.nullBitmap = make([]byte, nBitmap)
+	}
+
+	if c.elemBuf != nil && len(c.elemBuf) >= typeSize {
+		c.elemBuf = c.elemBuf[:typeSize]
+	} else {
+		c.elemBuf = make([]byte, typeSize)
+	}
+
+	c.length = length
+}
+
+// SetNull sets the rowIdx to null.
+func (c *Column) SetNull(rowIdx int) {
+	if !c.IsNull(rowIdx) {
+		c.nullBitmap[rowIdx>>3] &= ^(1 << uint(rowIdx&7))
+	}
+}
+
+// NullCount returns the number of nulls in this Column.
+func (c *Column) NullCount() int {
+	cnt := 0
+	i := 0
+	for ; i+8 <= c.length; i += 8 {
+		// 0 is null and 1 is not null
+		cnt += 8 - bits.OnesCount8(uint8(c.nullBitmap[i>>3]))
+	}
+	for ; i < c.length; i++ {
+		if c.IsNull(i) {
+			cnt++
+		}
+	}
+	return cnt
+}
+
+// PreAllocInt64 allocates space for an int64 slice and resets all slots to null.
+func (c *Column) PreAllocInt64(length int) {
```

Comment:
unused function?

Reply:
It will be actually used by expressions in the next PR.

For example:
```
c.PreAllocInt64(1024)
i64s := c.Int64s()
for i := 0; i < 1024; i++ {
    i64s[i] = calculateRow(i)
}
```